src/PotentialSimulation/ScalarPotential.jl

```diff
@@ -1,4 +1,8 @@
-const ScalarPotential{T, N, S} = Union{ElectricPotential{T, N, S}, WeightingPotential{T, N, S}, PointTypes{T, N, S}, EffectiveChargeDensity{T, N, S}}
+const ScalarPotential{T, N, S, AT} = Union{ElectricPotential{T, N, S, AT}, WeightingPotential{T, N, S, AT}, PointTypes{T, N, S, AT}, EffectiveChargeDensity{T, N, S, AT}}
```

Comment:
Should we introduce a type here?
`ElectricPotential <: ScalarPotential <: AbstractArray`?

Reply:
I am actually thinking whether we should remove `ElectricPotential`, `WeightingPotential`, `PointTypes`, `EffectiveChargeDensity` and just define `ScalarPotential` and parse the information what is stored in that object in some other why. But, this would require changes at several places.